redbeams/src/main/java/com/sequenceiq/redbeams/service/rotation/RedbeamsRotationService.java

```diff
@@ -0,0 +1,48 @@
+package com.sequenceiq.redbeams.service.rotation;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.cloudbreak.auth.crn.Crn;
+import com.sequenceiq.cloudbreak.common.exception.CloudbreakServiceException;
+import com.sequenceiq.cloudbreak.rotation.secret.RotationFlowExecutionType;
+import com.sequenceiq.cloudbreak.rotation.secret.SecretType;
+import com.sequenceiq.cloudbreak.rotation.secret.type.RedbeamsSecretType;
+import com.sequenceiq.flow.api.model.FlowIdentifier;
+import com.sequenceiq.redbeams.flow.RedbeamsFlowManager;
+
+@Service
+public class RedbeamsRotationService {
+
+    @Inject
+    private RedbeamsFlowManager redbeamsFlowManager;
+
+    @Inject
+    private EntitlementService entitlementService;
+
+    public FlowIdentifier rotateSecrets(String resourceCrn, List<String> secrets, RotationFlowExecutionType executionType) {
+        if (entitlementService.isSecretRotationEnabled(Crn.fromString(resourceCrn).getAccountId())) {
+            if (!secrets.stream().anyMatch(secret -> Collections.frequency(secrets, secret) > 1)) {
+                List<SecretType> secretTypes = secrets.stream()
+                        .filter(secret -> Arrays.stream(RedbeamsSecretType.values())
+                                .map(Enum::name)
+                                .toList()
+                                .contains(secret))
+                        .map(RedbeamsSecretType::valueOf)
+                        .collect(Collectors.toList());
```

Comment:
maybe we could create some util class for these 2 conditions in framework since StackOperationService

Reply:
Good idea